### PR TITLE
Added snapping

### PR DESCRIPTION
**Pull request by arnasbr**
Adding `snapping` support, related to https://github.com/traveltime-dev/traveltime-python-sdk/issues/121.

### CODE REVIEW

File: traveltimepy/dto/common.py [hunk 1]
```diff
@@ -123,11 +123,23 @@ class Property(str, Enum):
...
+class SnappingAcceptRoads(str, Enum):
+    BOTH_DRIVABLE_AND_WALKABLE = "both_drivable_and_walkable"
+    ANY_DRIVABLE = "any_drivable"
+
+
+class Snapping(BaseModel):
+    penalty: Optional[SnappingPenalty] = SnappingPenalty.ENABLED
+    accept_roads: Optional[SnappingAcceptRoads] = (
+        SnappingAcceptRoads.BOTH_DRIVABLE_AND_WALKABLE
+    )
```
Comment (arnasbr) [inline traveltimepy/dto/common.py:140]: default values of `None` should work here too. This way it's easy to see the default value, but `None` wouldn't require changing if default value in api changes

File: README.md [hunk 2]
```diff
@@ -70,9 +70,7 @@ Given origin coordinates, find shapes of zones reachable within corresponding tr
 * search_range: Range - When enabled, range adds an arrival window to the arrival time, and results are returned for any
   journeys that arrive during this window.
 * level_of_detail: LevelOfDetail - When enabled, allows the user to specify how detailed the isochrones should be.
-* snap_penalty: SnapPenalty - When enabled, walking time and distance from the departure location to the nearest road, 
-  and from the nearest road to the arrival location, are added to the total travel time and distance of a journey. 
-  Enabled by default.
+* snapping: Snapping - Adjusts the process of looking up the nearest roads from the departure / arrival points.
```
Comment (pawel-wroniszewski) [inline README.md:73]: Maybe add an internal link to the Snapping section below. I would do it for all places where this param is mentioned.

Comment (arnasbr) [reply README.md:73]: tbh I could do it for every single param, but probably in a separate PR. For now, at the end of the README I have a section dedicated to more complex params like this, and they point to the docs

Comment (pawel-wroniszewski) [reply README.md:73]: Yeah I wanted to add a link to this exact section

Comment (arnasbr) [reply README.md:73]: alright, in the next PR I'll go through and add hyperlinks to our docs (at least for the more complex params, like `level_of_detail` and `snapping`)

Comment (arnasbr) [reply README.md:73]: don't wanna make this PR too big

**Review by danielnaumau — APPROVED** [review]

File: traveltimepy/dto/common.py [hunk 3]
```diff
@@ -143,8 +153,14 @@ class Range(BaseModel):
...
 
+class ScaleType(str, Enum):
+    SIMPLE = "simple"
+    SIMPLE_NUMERIC = "simple_numeric"
+    COARSE_GRID = "coarse_grid"
+
+
 class LevelOfDetail(BaseModel):
-    scale_type: Literal["simple", "simple_numeric", "coarse_grid"] = "simple"
+    scale_type: ScaleType = ScaleType.SIMPLE
```
[the remark this answers is not in the document]
Comment (danielnaumau) [reply traveltimepy/dto/common.py:165]: Literal is also kinda enum.

[the remark this answers is not in the document]
Comment (danielnaumau) [reply traveltimepy/dto/common.py:165]: but you can do it like this if you want

**Comment by arnasbr** [comment]
Decided not to change level_of_detail as that would introduce unnecessary breaking changes. Reverting that part of the PR and will merge

**Pull request merged**